core/src/main/scala/org/apache/spark/util/ThreadStackTrace.scala

```diff
@@ -24,4 +24,9 @@ private[spark] case class ThreadStackTrace(
   threadId: Long,
   threadName: String,
   threadState: Thread.State,
-  stackTrace: String)
+  stackTrace: String,
+  blockedByThreadId: Long,
```

Comment:
Probably cleaner that way. I was passing on what JMX returns, but I think you're right that we should translate to Scala expectations.